src/js/tools/type.js

```diff
@@ -38,13 +44,32 @@ define(function (require, exports, module) {
             firstLaunch = true;
             
         var selectHandler = function () {
+            // If this is set, means we didn't get to deselect the tool last time
+            if (_moveHandler) {
```

Comment:
@iwehrman This is best I can do, during error caused reset, deselect handler does not get called.